core/payment/src/api/invoices.rs

```diff
@@ -321,6 +321,16 @@ async fn accept_invoice(
         }
         Err(e) => return response::server_error(&e),
     };
+    // OK when invoice.amount is greater than or equal to agreement.amount_accepted
+    if &invoice.amount < &agreement.total_amount_accepted.0 {
+        let msg = format!(
+            "Invoice is smaller than agreement.total_amount_accepted. \
+            invoice_id={}, invoice_amount={}, total_amount_accepted={}",
+            &invoice_id, &invoice.amount, &agreement.total_amount_accepted
+        );
+        log::warn!("{}", msg);
+        return response::bad_request(&msg);
+    }
```

Comment:
Payment service shouldn't allow this invoice **to be even received**.

Reply:
should i still move this check "up" in the chain?

Reply:
I meant that this check should be performed somewhere `service::sent_invoice` so that an invoice with lower amount than `agreement.total_amount_accepted` is never added to the requestor's db.